model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -636,7 +636,10 @@ def model_output_to_completion_output(
                 )
             except Exception as e:
                 logger.exception(f"Error parsing text-generation-inference output {model_output}")
-                raise e
+                if 'generated_text' not in model_output:
+                    raise ObjectHasInvalidValueException(
```

Comment:
Hmm yeah I'm not sure it necessarily follows that a user passed a bad input, if we observe `if 'generated_text' not in model_output:`. It seems like we're doing this empirically instead of going against TGI's API (not sure if they document this behavior). Perhaps the original exception that gets thrown can tell us something? @yunfeng-scale thoughts?

Reply:
i don't think TGI's error behavior is documented, they just send back a string when error

can you change error handling for streaming as well?

Reply:
yeah I'll update the error handling for streaming as well

Reply:
The initial error message from TGI is a stringified json message: "{'error': 'Input validation error: `inputs` must have less than 2048 tokens. Given: 2687', 'error_type': 'validation'}". I could parse this error and return it to the user wrapped as an InputValidationError? Does that sound better? @yixu34 @yunfeng-scale

Reply:
Generally not ideal to try and parse strings, and it's instead preferable to rely on structured fields. But in the absence of that, think this is fine for now. It's at least more precise than inferring semantics like what's currently being done in this PR?

I'd be in favor of doing the parse, putting a TODO, and revisiting this.

Reply:
i think we could try to parse it as an error and return more precise content. the current error `Error parsing text-generation-inference output` does not really capture what the error is and might confuse user

Reply:
Okay added this logic for completion sync